[BUG] error out from readIdentifyHash if it did not read enough data

### General information

  * OS: Windows
  * Hypervisor: Hyper-V
  * Did you run `crc setup` before starting it (Yes/No)? No, I cannot get to that step
  * Running CRC on: Laptop

## CRC version

2.19.0

```bash
$ crc version
panic: runtime error: index out of range [7] with length 0

goroutine 1 [running]:
encoding/binary.littleEndian.Uint64(...)
        /usr/lib/golang/src/encoding/binary/binary.go:77
github.com/crc-org/crc/pkg/crc/segment.readIdentifyHash({0xc000c2a3c0?, 0x16?})
        /builddir/build/BUILD/crc-2.19.0/pkg/crc/segment/segment.go:142 +0xb4
github.com/crc-org/crc/pkg/crc/segment.newCustomClient(0xc0006678c0, {0x21f8640, 0x32040e0}, {0xc000c2a390?, 0x0?}, {0xc000c2a3c0, 0x27}, {0x1eabc69, 0x16})
        /builddir/build/BUILD/crc-2.19.0/pkg/crc/segment/segment.go:66 +0x1f3
github.com/crc-org/crc/pkg/crc/segment.NewClient(0x32090a0?, {0x21f8640, 0x32040e0})
        /builddir/build/BUILD/crc-2.19.0/pkg/crc/segment/segment.go:43 +0x147
github.com/crc-org/crc/cmd/crc/cmd.init.7()
        /builddir/build/BUILD/crc-2.19.0/cmd/crc/cmd/root.go:63 +0x150
```
  
## CRC status

N/A cannot run crc

```bash
# Put `crc status --log-level debug` output here
```

## CRC config

N/A cannot run crc

```bash
# Put `crc config view` output here
```

## Host Operating System
```bash
$ systeminfo

Host Name:                 PSCHULTZ-LAPTOP
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.19045 N/A Build 19045
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          pschultz@pobox.com
Registered Organization:   N/A
Product ID:                00330-80846-80580-AA719
Original Install Date:     8/9/2021, 09:13:09
System Boot Time:          5/30/2023, 09:42:17
System Manufacturer:       Eluktronics Inc.
System Model:              MAX-17
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: AMD64 Family 25 Model 80 Stepping 0 AuthenticAMD ~3301 Mhz
BIOS Version:              American Megatrends International, LLC. N.1.18ELU05, 5/21/2021
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume5
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-06:00) Central Time (US & Canada)
Total Physical Memory:     32,158 MB
Available Physical Memory: 20,157 MB
Virtual Memory: Max Size:  37,022 MB
Virtual Memory: Available: 19,835 MB
Virtual Memory: In Use:    17,187 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\PSCHULTZ-LAPTOP
Hotfix(s):                 29 Hotfix(s) Installed.
                           [01]: KB5026513
                           [02]: KB4562830
                           [03]: KB4570334
                           [04]: KB4577586
                           [05]: KB4580325
                           [06]: KB4586864
                           [07]: KB4598481
                           [08]: KB5000736
                           [09]: KB5003791
                           [10]: KB5012170
                           [11]: KB5015684
                           [12]: KB5008575
                           [13]: KB5026435
                           [14]: KB5006753
                           [15]: KB5007273
                           [16]: KB5011352
                           [17]: KB5011651
                           [18]: KB5014032
                           [19]: KB5014035
                           [20]: KB5014671
                           [21]: KB5015895
                           [22]: KB5016705
                           [23]: KB5018506
                           [24]: KB5020372
                           [25]: KB5022924
                           [26]: KB5023794
                           [27]: KB5025315
                           [28]: KB5026879
                           [29]: KB5005699
Network Card(s):           8 NIC(s) Installed.
                           [01]: Realtek Gaming 2.5GbE Family Controller
                                 Connection Name: Ethernet
                                 Status:          Media disconnected
                           [02]: Intel(R) Wi-Fi 6 AX200 160MHz
                                 Connection Name: Wi-Fi
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.68.1
                                 IP address(es)
                                 [01]: 192.168.68.59
                                 [02]: fe80::3e0c:655d:5452:fc9b
                           [03]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection
                                 Status:          Media disconnected
                           [04]: ZeroTier Virtual Port
                                 Connection Name: ZeroTier One [8056c2e21c329a03]
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 172.25.237.196
                                 [02]: fe80::71b7:4665:84c1:2d2e
                           [05]: DisplayLink Network Adapter NCM
                                 Connection Name: Ethernet 2
                                 Status:          Media disconnected
                           [06]: TAP-Windows Adapter V9
                                 Connection Name: OpenVPN TAP-Windows6
                                 Status:          Media disconnected
                           [07]: TAP-NordVPN Windows Adapter V9
                                 Connection Name: Ethernet 3
                                 Status:          Media disconnected
                           [08]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (WSL)
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 192.168.16.1
                                 [02]: fe80::5c69:f749:636:7147
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```

### Steps to reproduce

  1. Download Podman Desktop
  2. Enable the Red Hat OpenShift Local extension
  3. Install OpenShift Local

Note: The issue still persists if I install the OpenShift Local executable directly from the [website](https://console.redhat.com/openshift/create/local).

### Expected

crc works

### Actual

crc runs into a panic error

### Logs

N/A cannot run crc

Before gather the logs try following if that fix your issue
```bash
$ crc delete -f
$ crc cleanup
$ crc setup
$ crc start --log-level debug
```

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.

--- a/pkg/crc/segment/segment.go
+++ b/pkg/crc/segment/segment.go
@@ -139,6 +139,10 @@ func readIdentifyHash(identifyHashPath string) (uint64, error) {
 		return 0, err
 	}
 
+	if length := len(cachedHashBytes); length != 8 {
+		return 0, fmt.Errorf("Got %d bytes hash, expected 8", length)
+	}
+
 	return binary.LittleEndian.Uint64(cachedHashBytes), nil
 }
 

--- a/pkg/crc/segment/segment.go
+++ b/pkg/crc/segment/segment.go
@@ -173,7 +173,9 @@ func (c *Client) upload(action string, a analytics.Properties) error {
 			return err
 		}
 		c.identifyHash = hash
-		_ = writeIdentifyHash(c)
+		if err := writeIdentifyHash(c); err != nil {
+			return fmt.Errorf("failed to write new identify hash to file: %w", err)
+		}
 	}
 
 	return c.segmentClient.Enqueue(analytics.Track{

